stan/math/rev/fun/fma.hpp

```diff
@@ -233,19 +171,243 @@ inline var fma(Ta&& a, Tb&& b, const var& c) {
  *
  * The partial derivatives are
  *
- * \f$\frac{\partial}{\partial y} (c * y) + z = c\f$, and
+ * \f$\frac{\partial}{\partial y} (x * y) + z = x\f$, and
  *
- * \f$\frac{\partial}{\partial z} (c * y) + z = 1\f$.
+ * \f$\frac{\partial}{\partial z} (x * y) + z = 1\f$.
  *
  * @tparam Ta type of the first multiplicand
- * @param a First multiplicand.
- * @param b Second multiplicand.
- * @param c Summand.
+ * @param x First multiplicand.
+ * @param y Second multiplicand.
+ * @param z Summand.
  * @return Product of the multiplicands plus the summand, ($a * $b) + $c.
  */
-template <typename Ta, typename = require_arithmetic_t<Ta>>
-inline var fma(Ta&& a, const var& b, const var& c) {
-  return var(new internal::fma_vdv_vari(b.vi_, a, c.vi_));  // a-b symmetry
+template <typename Ta, require_arithmetic_t<Ta>* = nullptr>
+inline var fma(Ta&& x, const var& y, const var& z) {
```

Comment:
Ooof I let all these &&s escape. If I remember I'm gonna switch these to const refs in the future, but it's fine for now.